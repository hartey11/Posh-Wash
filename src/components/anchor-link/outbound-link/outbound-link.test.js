import React from "react"
import { render, cleanup, fireEvent } from "@testing-library/react"
import OutboundLink from "./outbound-link.component"

afterEach(cleanup)

const clickEvent = jest.fn()
const title = "Some Title"
const href = "https://www.bbc.co.uk/this should encode correctly"

describe("OutboundLink Component", () => {
  it("should render correctly", () => {
    const { asFragment } = render(
      <OutboundLink isBrowser title={title} href={href}>
        Test
      </OutboundLink>
    )
    expect(asFragment()).toMatchSnapshot()
  })

  it("should render without target and rel attributes if href is not http", () => {
    const { asFragment } = render(
      <OutboundLink isBrowser title={title} href="mailto:name@email.com">
        Test
      </OutboundLink>
    )
    expect(asFragment()).toMatchSnapshot()
  })

  it('should render without rel attributes if target is not "_blank"', () => {
    const { asFragment } = render(
      <OutboundLink isBrowser target="_self" title={title} href={href}>
        Test
      </OutboundLink>
    )
    expect(asFragment()).toMatchSnapshot()
  })

  it("should render with a clickEvent", () => {
    const { asFragment } = render(
      <OutboundLink clickEvent={clickEvent} isBrowser title={title} href={href}>
        Test
      </OutboundLink>
    )
    expect(asFragment()).toMatchSnapshot()
  })

  it("should trigger the click event correctly", () => {
    global.encodeURI = jest.fn()

    const { getByText } = render(
      <OutboundLink clickEvent={clickEvent} isBrowser title={title} href={href}>
        Click Me
      </OutboundLink>
    )

    fireEvent.click(getByText("Click Me"))

    expect(clickEvent).toHaveBeenCalled()
  })
})
