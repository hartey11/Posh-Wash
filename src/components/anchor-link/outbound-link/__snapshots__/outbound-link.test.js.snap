// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`OutboundLink Component should render correctly 1`] = `
<DocumentFragment>
  <a
    href="https://www.bbc.co.uk/this%20should%20encode%20correctly"
    rel="noopener noreferrer"
    target="_blank"
    title="Some Title"
  >
    Test
  </a>
</DocumentFragment>
`;

exports[`OutboundLink Component should render with a clickEvent 1`] = `
<DocumentFragment>
  <a
    href="https://www.bbc.co.uk/this%20should%20encode%20correctly"
    rel="noopener noreferrer"
    target="_blank"
    title="Some Title"
  >
    Test
  </a>
</DocumentFragment>
`;

exports[`OutboundLink Component should render without rel attributes if target is not "_blank" 1`] = `
<DocumentFragment>
  <a
    href="https://www.bbc.co.uk/this%20should%20encode%20correctly"
    target="_self"
    title="Some Title"
  >
    Test
  </a>
</DocumentFragment>
`;

exports[`OutboundLink Component should render without target and rel attributes if href is not http 1`] = `
<DocumentFragment>
  <a
    href="mailto:name@email.com"
    title="Some Title"
  >
    Test
  </a>
</DocumentFragment>
`;
